RealifeTech-CoreSDK/APILayer/DiskCache/DiskCache.swift

```diff
@@ -8,112 +8,169 @@
+            let text = try String(contentsOf: fileUrl, encoding: .utf8)
+            let expired = isFileExpired(text)
             if !includeExpired && expired {
                 return (file: nil, expired: true)
             }
-            return (file: fileComponent(text: text, key: .file), expired: expired)
+            let file = getFileComponent(by: .file, from: text)
+            return (file: file, expired: expired)
         } catch {
             return (file: nil, expired: false)
         }
     }
 
-    static func readItems(withBaseFileName baseFileName: String) -> [String] {
-        guard let dir = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else { return [] }
-        guard let files = try? FileManager.default.contentsOfDirectory(at: dir, includingPropertiesForKeys: nil, options: .skipsHiddenFiles) else { return [] }
+    public func readItems(with baseFileName: String) -> [String] {
+        guard let files = try? getAllFilesUrl() else { return [] }
         return files
             .filter { return $0.lastPathComponent.starts(with: baseFileName) }
             .map { try? String(contentsOf: $0.absoluteURL, encoding: .utf8) }
             .compactMap { $0 }
-            .map { return fileComponent(text: $0, key: .file) }
+            .map { return getFileComponent(by: .file, from: $0) }
             .compactMap { $0 }
     }
 
-    public static func clear(deletionStrategy: DeletionStrategy, completion: (() -> Void)? = nil) {
-        guard let dir = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else { return }
-        DispatchQueue.global(qos: .background).async {
-            guard let files = try? FileManager.default.contentsOfDirectory(at: dir, includingPropertiesForKeys: nil, options: .skipsHiddenFiles) else { return }
-            let fullExtension = deletionStrategy == .privateOnly ? "\(DiskCache.privateIndicator)\(DiskCache.fileExtension)" : "\(DiskCache.fileExtension)"
-            for file in files.filter({ $0.absoluteString.contains(fullExtension) }) {
-                var shouldDelete = false
-                switch deletionStrategy {
-                case .all, .privateOnly: shouldDelete = true
-                case .allNonPrivate: shouldDelete = !file.absoluteString.contains("\(DiskCache.privateIndicator)")
-                case .outdatedOnly:
-                    guard let text = try? String(contentsOf: file.absoluteURL, encoding: .utf8) else { continue }
-                    shouldDelete = fileExpired(text: text, andAlsoOutdated: true)
+    public func deleteItem(with fileName: String) {
+        guard let fileUrl = try? getFileUrl(fileName) else { return }
+        try? fileManager.removeItem(at: fileUrl)
+    }
+
+    public func clearItems(deletionStrategy: DiskCacheDeletionStrategy, completion: (() -> Void)?) {
+        guard let files = try? getAllFilesUrl() else { return }
+        let fullExtension = deletionStrategy == .privateOnly ?
```

Comment:
What do you think of the following styling @mickeylee25 where we have the question mark and the : on the line with the value? I prefer this as the boolean expression is on the first line and the value associated with each statement are on their respective lines.

```
let fullExtension = deletionStrategy == .privateOnly 
    ? "\(Self.privateIndicator)\(Self.fileExtension)"
    : "\(Self.fileExtension)"
```

Reply:
I don't have objections to this. The format is used in another React project so I keep using here.